test/runtests.jl

```diff
@@ -53,8 +53,20 @@ ia2 = copy(ia)
 @test ia2[:] == ["cat", "dog", "mouse"]
 @test findfirst(ia2, "cat") == 1
 
+let ia = IndexedArray(["cat", "dog", "mouse", "human"]), original = copy(ia)
+  swap!(ia, 2, 3)
+
+  @test findfirst(ia, "mouse") == 2
+  @test findfirst(original, "mouse") == 3
+
+  @test findfirst(ia, "dog") == 3
+  @test findfirst(original, "dog") == 2
+end
+
+
```

Comment:
Thanks!  Can you switch to a four-space indent and also get rid of the added newline at the end of the file?

I think it would also be good to have a test for when `to == from`.  Perhaps the function itself should special-case for this and return immediately as well.